configure

```diff
@@ -538,13 +542,12 @@ detect_resolv_dynamic()
 	esac
 	TMPFILE=$(mktemp /tmp/bitlbee-configure.XXXXXX)
 	ret=1
-	echo "$RESOLV_TESTCODE" | $CC -o $TMPFILE -x c - $LIBRESOLV >/dev/null 2>/dev/null
-	if [ "$?" = "0" ]; then
+	if echo "$RESOLV_TESTCODE" | $CC -o "$TMPFILE" -x c - $LIBRESOLV >/dev/null 2>/dev/null; then
```

Comment:
Why this change?

Reply:
This is [SC2181](https://www.shellcheck.net/wiki/SC2181) and should avoid some redundancy.

Reply:
In this particular instance, I personally find the original code easier to read spread out over two lines.

Reply:
Okay, while that's likely a matter of taste, I can change it back. And all occurrences of changed `$…TESTCODE`, I guess?

Reply:
If you let me know, I'm happy to update my pull request.

Reply:
Yeah, can you perhaps revert this for now - in the interest of being conservative? I think the line is long enough to be split up.